waveform_base.py

```diff
@@ -818,6 +818,29 @@ def interpolate(self, tprime):
         return W
 
 
+    @waveform_alterations
+    def conjugate(self):
+        """Complex conjugate the waveform data
```

Comment:
Good point that there is something subtle here. This method gets inherited by both WaveformModes and WaveformGrid, and has a different action in the two cases: either conjugating w(u,theta,phi) or w_lm(u). Both operations may be desired, but they are distinct and should not be confused. Maybe the name is bad for WaveformModes. Maybe WaveformModes also needs a function which applies conjugation, then sends the mode (l,m) mode to (l,-m) and multiplies by the appropriate parity?

Reply:
Is this function actually needed?  I'm a little worried because it breaks certain promises made by the code, related to the fact that a Waveform is supposed to know what its data type is (which is how it knows its spin weight, boost weight, etc.).  So if you really want this function, we'd need to add each data type's conjugate to the list...

Reply:
@moble good point, I didn't think more deeply about that fact that a conjugated waveform should have opposite spin weight. I now think this is too subtle an issue to be handled by a function at the level of WaveformBase and will remove it (since also it is not used by the rest of this PR)

Reply:
Removed.